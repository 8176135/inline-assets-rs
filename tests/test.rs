extern crate inline_assets;

#[test]
fn basic() {
    assert_eq!(inline_assets::inline_file("examples/resources/listener_screen.html", true).unwrap(), include_str!("listener_screen.compiled.html"));
}
#[test]
fn css_import_and_http() {
    assert_eq!(inline_assets::inline_file("examples/resources/listener_screen_css_import.html", true).unwrap(), include_str!("listener_screen_css_import.compiled.html"));
}
