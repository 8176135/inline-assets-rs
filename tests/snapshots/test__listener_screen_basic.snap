---
created: "2019-08-26T23:00:57.750359200Z"
creator: insta@0.10.1
source: tests/test.rs
expression: "inline_assets::inline_file(\"examples/resources/listener_screen.html\",\n                           inline_assets::Config{remove_new_lines: false,\n                                                 inline_fonts:\n                                                     false,}).unwrap()"
---
"<html><head>\n    <style>@font-face {\n    font-family: \'source-sans\';\n    src: url(another_sub_dir/sourcesanspro-regular-webfont.woff2) format(\"woff\");\n}\n\nbody {\n    font-family: \"source-sans\", serif;\n    font-size: 10pt;\n    -webkit-user-select: none;\n    -moz-user-select: none;\n    -ms-user-select: none;\n    user-select: none;\n\n    background-color: #CCC;\n    color: #222;\n}\n\np {\n    cursor: default;\n}\n\n#port_num {\n    width: 50px;\n}\n\nbutton {\n    width: auto;\n    left: 5px;\n    right: 5px;\n    bottom: 5px;\n    position: absolute;\n    border: none;\n    background-color: #7d81ff;\n    color: #EEE;\n}\n\nbutton:hover {\n    background-color: #a9a7ff\n}\n\ninput, label {\n    display: block;\n}\n\n#is_connected {\n    margin-left: 20px;\n    font-family: Consolas, monospace;\n}</style>\n</head>\n<body>\n<p style=\"overflow:hidden; max-height: 43px\"><b>Status: </b> <span id=\"is_connected\">Not listening</span></p>\n<form>\n    <label for=\"port_num\">Port:</label>\n    <input id=\"port_num\" max=\"65500\" min=\"1\" name=\"port_num\" placeholder=\"13111\" type=\"number\">\n    <br><br>\n    <button onclick=\"external.invoke(\'listen \' + valOrPlaceholder(document.getElementById(\'port_num\')))\" type=\"submit\">Listen</button>\n</form>\n<script>function valOrPlaceholder(item) {\n    if (item.value) {\n        return item.value;\n    } else {\n        return item.placeholder;\n    }\n}</script>\n<script type=\"text/javascript\">\n    function showConnectedMsg(e) {\n        document.getElementById(\'is_connected\').innerHTML = e;\n    }\n</script>\n\n</body></html>"
